neuraxle/pipeline.py

```diff
@@ -249,6 +320,20 @@ def _find_starting_step_index(self, data_container: DataContainer) -> int:
 
         return index_latest_checkpoint
 
+    def _load_pipeline_cache(self, pipeline: 'Pipeline', new_starting_step_index):
+        steps_before_checkpoint_changed = False
+        for index, (step_name, step) in enumerate(self[:new_starting_step_index + 1]):
+            bytecode_current_step = dis.dis(step)
+            bytecode_cached_step = dis.dis(pipeline.steps[index])
+            if bytecode_current_step != bytecode_cached_step:
+                steps_before_checkpoint_changed = True
```

Comment:
extract method perhaps. To name what is done and have less variables in head.

Reply:
see next commit ;)